spec/system/users_show_spec.rb

```diff
@@ -1,72 +1,79 @@
 require 'rails_helper'
 
 RSpec.describe 'Users#show', type: :feature, js: true do
-  let(:user) { User.create!(name: 'Lilly', photo: 'https://portraits.ai', bio: 'Teacher from Poland.') }
-  let!(:posts) do
-    [
-      Post.create(author: user, title: 'Post 1', text: 'text 1'),
-      Post.create(author: user, title: 'Post 2', text: 'text 2'),
-      Post.create(author: user, title: 'Post 3', text: 'text 3'),
-      Post.create(author: user, title: 'Post 4', text: 'text 4')
-    ]
-  end
   before(:all) do
     User.destroy_all
     Post.destroy_all
     Comment.destroy_all
+
+    @lilly = User.create!(name: 'Lilly', photo: 'https://portraits.ai', bio: 'Teacher from Poland.', posts_counter: 0)
+    @post = Post.create!(title: 'First Post', text: 'This is my first post', author: @lilly, comments_counter: 0,
+                         likes_counter: 0)
+    @tom = User.create!(name: 'Tom', photo: 'https://portraits.ai', bio: 'Teacher from Mexico.', posts_counter: 0)
+    @users = User.all
   end
+
   # I can see the user's profile picture.
   it 'displays the user profile picture' do
     visit '/users'
-    click_on(user.name)
-    expect(page).to have_css("img[src*='#{user.photo}']")
+    click_on 'Lilly'
+    expect(page).to have_css("img[src*='#{@lilly.photo}']")
   end
+
   # I can see the user's username.
   it 'displays the user username' do
     visit '/users'
-    click_on(user.name)
-    expect(page).to have_content(user.name.to_s)
+    click_on 'Lilly'
+    expect(page).to have_content(@lilly.name.to_s)
   end
+
   # I can see the number of posts the user has written.
   it 'displays the number of posts the user has written' do
     visit '/users'
-    click_on(user.name)
+    click_on 'Lilly'
     expect(page).to have_content('1')
   end
+
   # I can see the user's bio.
   it 'displays the user bio' do
     visit '/users'
-    click_on(user.name)
+    click_on 'Lilly'
     expect(page).to have_content('Teacher from Poland.')
   end
+
   # I can see the user's first 3 posts.
   it 'displays the user first 3 posts' do
     visit '/users'
-    click_on(user.name)
-    user.three_recent_posts.each do |_post|
-      expect(page).to have_content('post')
-    end
+    click_on 'Lilly'
+    expect(page).to have_content('First Post')
```

Comment:
- [ ] @Team Great work adding tests for the users' index page, however on this test, you are supposed to check if the `three_recent_posts` method is working correctly, that given more than three posts it will display three recent posts.(check the example below)

` RSpec.describe "Posts", type: :request do`
 ` before do`
    `@post1 = Post.create(title: "Post 1", body: "Post 1 body")`
    `@post2 = Post.create(title: "Post 2", body: "Post 2 body")`
    `@post3 = Post.create(title: "Post 3", body: "Post 3 body")`
    `@post4 = Post.create(title: "Post 4", body: "Post 4 body")`
 `end`
  `describe "GET /index" do`
    `it "displays 3 most recent posts" do`
      `get posts_path`
      `expect(response).to have_http_status(:success)`
      `# Check that the 3 most recent posts are displayed`
      `expect(response.body).to include(@post2.title)`
     ` expect(response.body).to include(@post3.title)`
      `expect(response.body).to include(@post4.title)`
      `# Check that the oldest post is not displayed`
      `expect(response.body).not_to include(@post1.title)`
    `end`
  `end`
`end`